src/methods.jl

```diff
@@ -10,318 +19,453 @@ derivatives.
+function (m::FiniteDifferenceMethod)(
+    f::Function,
+    x::T,
+    h
+) where T<:AbstractFloat
+    return sum(
+        i -> convert(T, m.coefs[i]) * f(T(x + h * m.grid[i])),
+        eachindex(m.grid)
+    ) / h^m.q
 end
+# Handle arguments that are not floats. Assume that converting to float is desired.
+(m::FiniteDifferenceMethod)(f::Function, x::T, h) where T<:Real = m(f, float(x), h)
 
-# Check the method and derivative orders for consistency
+# Check the method and derivative orders for consistency.
 function _check_p_q(p::Integer, q::Integer)
-    q >= 0 || throw(ArgumentError("order of derivative must be nonnegative"))
+    q >= 0 || throw(ArgumentError("order of derivative must be non-negative"))
     q < p || throw(ArgumentError("order of the method must be strictly greater than that " *
                                  "of the derivative"))
-    # Check whether the method can be computed. We require the factorial of the
-    # method order to be computable with regular `Int`s, but `factorial` will overflow
-    # after 20, so 20 is the largest we can allow.
+    # Check whether the method can be computed. We require the factorial of the method order
+    # to be computable with regular `Int`s, but `factorial` will after 20, so 20 is the
+    # largest we can allow.
     p > 20 && throw(ArgumentError("order of the method is too large to be computed"))
     return
 end
 
-# Compute coefficients for the method
+const _COEFFS_CACHE = Dict{Tuple{AbstractVector{<:Real}, Integer}, Vector{Float64}}()
 
-const _COEFFS_CACHE = Dict{Tuple{AbstractVector{<:Real}, Integer, Integer}, Vector{Float64}}()
-function _coefs(grid::AbstractVector{<:Real}, p::Integer, q::Integer)
-    return get!(_COEFFS_CACHE, (grid, p, q)) do
-        # For high precision on the \ we use Rational, and to prevent overflows we use Int128
-        # At the end we go to Float64 for fast floating point math (rather than rational math)
+# Compute coefficients for the method and cache the result.
+function _coefs(grid::AbstractVector{<:Real}, q::Integer)
+    return get!(_COEFFS_CACHE, (grid, q)) do
+        p = length(grid)
+        # For high precision on the `\`, we use `Rational`, and to prevent overflows we use
+        # `Int128`. At the end we go to `Float64` for fast floating point math, rather than
+        # rational math.
         C = [Rational{Int128}(g^i) for i in 0:(p - 1), g in grid]
         x = zeros(Rational{Int128}, p)
         x[q + 1] = factorial(q)
         return Float64.(C \ x)
     end
 end
 
+# Estimate the bound on the derivative by amplifying the ∞-norm.
+function _make_default_bound_estimator(; condition::Int=DEFAULT_CONDITION)
+    return (f, x) -> condition * maximum(abs.(f(x)))
+end
 
-# Estimate the bound on the function value and its derivatives at a point
-_estimate_bound(x, cond) = add_tiny(cond * maximum(abs, x))
+function Base.show(io::IO, x::FiniteDifferenceMethod)
```

Comment:
the 2 arg show should not return something that is not valid julia code.
Since it is what 1arg `repr` hits.

Instread we want:
```suggestion
function Base.show(io::IO, m::MIME"text/plain", t::Template)
```

I think.
Which just controls the plain text display used by the REPL etc

Reply:
Gotcha! Thanks.

I'll change it to

```suggestion
function Base.show(io::IO, m::MIME"text/plain", x::FiniteDifferenceMethod)
```